test/replication-luatest/gh_5158_invalid_recovery_from_xlog_test.lua

```diff
@@ -0,0 +1,78 @@
+local t = require('luatest')
+local cluster = require('test.luatest_helpers.cluster')
+local server = require('test.luatest_helpers.server')
+local fio = require('fio')
+local log = require('log')
+
+local g = t.group('gh_5158')
+
+g.before_each(function(cg)
+    cg.cluster = cluster:new({})
+    local box_cfg = {
+        checkpoint_count = 5,
+        log_level = 6,
+    }
+
+    cg.master = cg.cluster:build_server({alias = 'master', box_cfg = box_cfg})
```

Comment:
You may use `build_and_add_server` instead of `build_server` + `add_server`.

Reply:
Done